text/0635-mediaconvert-l2-construct.md

```diff
@@ -0,0 +1,258 @@
+                Destination: 's3://amzn-s3-demo-destination-bucket/'
+              }
+            },
+            Outputs: [
+              {
+                ContainerSettings: {
+                  Container: 'MP4'
+                },
+                VideoDescription: {
+                  CodecSettings: {
+                    Codec: 'H_264',
+                    H264Settings: {
+                      MaxBitrate: 1000,
+                      RateControlMode: 'QVBR',
+                      SceneChangeDetect: 'TRANSITION_DETECTION'
+                    }
+                  }
+                },
+                AudioDescriptions: [
+                  {
+                    CodecSettings: {
+                      Codec: 'AAC',
+                      AacSettings: {
+                        Bitrate: 96000,
+                        CodingMode: 'CODING_MODE_2_0',
+                        SampleRate: 48000
+                      }
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    };
+
+    // Define the custom MediaConvert Job Template
+    const customJobTemplate = new JobTemplate(this, 'CustomJobTemplate', jobTemplateProps);
+
+In this example, the ``customJobTemplate`` is configured with specific output groups and encoding settings. By defining 
+these properties, you can create a Job Template that aligns precisely with your media processing requirements, ensuring 
+consistent and optimized workflows across your jobs.
+
+The System Job Templates are a collection of pre-configured and managed JobTemplates provided by AWS Elemental MediaConvert. 
+These templates are designed to cater to common media processing use cases and are set up with best practices in mind. 
+By utilizing these System Job Templates, you can easily configure your MediaConvert jobs by simply referencing the template's 
+name. The necessary settings and configurations will be automatically applied, ensuring that your media processing workflows 
+are optimized and aligned with recommended standards.
+
+    const genericMp4Hev1AvcAacSdrQvbJobTemplate = new JobTemplate(this, 'GenericMp4Hev1AvcAacSdrQvbJobTemplate', {
+        name: "GenericMp4Hev1AvcAacSdrQvbJobTemplate"
+    });
+
+In this example, the ``genericMp4Hev1AvcAacSdrQvbJobTemplate`` is a predefined System Job Template. By specifying its name,
+you automatically leverage a set of optimized settings tailored for a specific media conversion scenario. This approach
+simplifies the setup process, allowing you to focus on other aspects of your media processing pipeline.
+
+#### Presets
+
+To create a Custom Preset using an L2 Construct, you define the ``presetProps`` object, which includes all the necessary 
+settings for your media processing tasks. This object is then passed to the ``Preset`` construct to create the custom preset.
+
+    const presetProps: PresetProps = {
+        name: 'CustomPreset',
+        settings: {
```

Comment:
Will the objects in this `settings` have types defined? 

Same question to the `settings` in `JobTemplateProps`.


And in these 2 settings objects, are all fields required?

Reply:
Yes, my idea is to have strongly-typed fields for all of these interfaces by importing our SDK. Discussed offline with @moelasmar and @samson-keung and pending decision on whether we can use SDK inside CDK. 

To your second que: **no**, not all fields are required. I plan to add respective interfaces (with fields marked as required/optional) in my next commit.